conda/cli/main_run.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import json
+import os
+from os.path import abspath, join
+import sys
+from tempfile import NamedTemporaryFile
+
+from ..base.context import context
+from ..common.compat import ensure_binary, iteritems, on_win
+from ..gateways.disk.delete import rm_rf
+from ..gateways.subprocess import subprocess_call
+
+
+def get_activated_env_vars():
+    env_location = context.target_prefix
+    if on_win:
+        env_var_map = _get_activated_env_vars_win(env_location)
+    else:
+        env_var_map = _get_activated_env_vars_unix(env_location)
+    env_var_map = {str(k): str(v) for k, v in iteritems(env_var_map)}
+    return env_var_map
+
+
+def _get_activated_env_vars_win(env_location):
+    try:
+        conda_bat = os.environ["CONDA_BAT"]
+    except KeyError:
+        conda_bat = abspath(join(sys.prefix, 'condacmd', 'conda.bat'))
+
+    temp_path = None
+    try:
+        with NamedTemporaryFile('w+b', suffix='.bat', delete=False) as tf:
+            temp_path = tf.name
+            tf.write(ensure_binary(
+                "@%CONDA_PYTHON_EXE% -c \"import os, json; print(json.dumps(dict(os.environ)))\""
+            ))
+        cmd_builder = [
+            "cmd.exe /K \"",
```

Comment:
Instead of `cmd.exe /K`, maybe this could use `/C` instead? Would it make sense to use `COMSPEC`?
If yes to both, maybe the logic in https://github.com/conda/conda/blob/616cf994f71db564047dd420216d2dbcb59010f3/conda/core/link.py#L776-L784 should be put into a function that can be reused here?